setup.py

```diff
@@ -47,7 +47,7 @@
         "setuptools>=40.6.0",
         "wheel",
         "cloudpickle>=1.6.0,<=2.0.0",
-        f"snowflake-connector-python>={CONNECTOR_DEPENDENCY_VERSION}",
+        f"snowflake-connector-python>={CONNECTOR_DEPENDENCY_VERSION}, <3.0.0",
```

Comment:
If you add a cap here, you should also have it for "pandas" and "secure-local-storage" options. But probably we don't need it because we have regression tests against latest connector branch and snowpark branch.

Reply:
I wonder if this is needed? I could imagine it helps prevent compatibility issues given this is a major version bump, but just curious if there are other motivations.

Reply:
Usually for library dependency we should add a cap and regard the next major version as having break changes.

Reply:
"pandas" and "secure-local-storage" already have caps defined in python connector so we don't need here?

pandas =
    pandas>=1.0.0,<1.5.0
    pyarrow>=8.0.0,<8.1.0
secure-local-storage =
    keyring!=16.1.0,<24.0.0

Reply:
I think what Jianzhun mean is several lines below: `f"snowflake-connector-python[pandas]>={CONNECTOR_DEPENDENCY_VERSION}"`